### PR TITLE
[improve] Refactor file cache and Improve the file cache strategy

### PR DESCRIPTION
**Pull request by Lchangliang**
# Proposed changes

Issue Number: close #xxx

## Problem summary

1. Refactor file cache. Before refactor, the file cache config format is "[{"path":"/path/to/file_cache","normal":21474836480,"persistent":10737418240,"query_limit":10737418240}]" and now change to "[{"path":"/mnt/disk3/selectdb_cloud/file_cache","total_size":21474836480,"query_limit":10737418240}]". It will be simpler than before.
2. Support more strategy. Support file cache priority. The file cache will have three queue,  name as 'index'/'normal'/'disposable'. We can avoid that the higher priority data is eliminate by the lower priority data.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Lchangliang** [comment]
run buildall

File: be/src/olap/options.cpp [hunk 1]
```diff
@@ -174,45 +173,59 @@ Status parse_conf_cache_paths(const std::string& config_path, std::vector<CacheP
...
-
-    settings.max_elements = std::max<size_t>(
-            normal_bytes / config::file_cache_max_file_segment_size, settings.max_elements);
-    settings.persistent_max_elements =
-            std::max<size_t>(persistent_bytes / config::file_cache_max_file_segment_size,
-                             settings.persistent_max_elements);
     settings.max_query_cache_size = query_limit_bytes;
+    size_t per_size = settings.total_size / io::percentage[3];
+    settings.disposable_queue_size = per_size * io::percentage[1];
+    settings.disposable_queue_elements =
```
[the remark this answers is not in the document]
Comment (Lchangliang) [reply be/src/olap/options.cpp:215]: I will remove it in future optimizations.

**Comment by Lchangliang** [comment]
run buildall

File: be/src/io/cache/block/block_file_cache.cpp [hunk 2]
```diff
@@ -51,16 +51,34 @@ IFileCache::Key IFileCache::hash(const std::string& path) {
...
+        return "_idx";
+    case CacheType::DISPOSABLE:
+        return "_disposable";
+    case CacheType::NORMAL:
+        return "";
+    }
+    return "";
+}
+
+CacheType IFileCache::string_to_cache_type(const std::string& str) {
```
Comment (AshinGau) [inline be/src/io/cache/block/block_file_cache.cpp:66]: Why does only match the first character?

**Comment by AshinGau** [comment]
LGTM

File: be/src/io/cache/block/block_file_cache.cpp (hunk 2 above)
Comment (Lchangliang) [reply be/src/io/cache/block/block_file_cache.cpp:66]: Only the first character is needed to distinguish the type. It is faster than comparing with string.

**Comment by Lchangliang** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**